auacm/app/modules/problem_manager/views.py

```diff
@@ -1,89 +1,341 @@
+    except KeyError as err:
+        return serve_error('Request header not found: ' + err[0],
+                           response_code=400)
+
+    # Commit everything to the database
+    pid = problem.commit_to_session()
+    problem_data.pid = pid
+    problem_data.commit_to_session()
+    for case in sample_cases:
+        case.pid = pid
+        case.commit_to_session()
+
+    # Store the judge data
+    directory = os.path.join(app.config['DATA_FOLDER'],
+                             'problems', str(problem.pid))
+    in_file.extractall(directory)
+    out_file.extractall(directory)
+    os.mkdir(os.path.join(directory, 'test'))
+    sol_file.save(os.path.join(directory, 'test', sol_file.filename))
+
+    return serve_response({
+        'success': True,
+        'name': problem.name,
+        'shortname': problem.shortname,
+        'description': problem_data.description,
+        'input_desc': problem_data.input_desc,
+        'output_desc': problem_data.output_desc,
+        'sample_cases': cases,
+        'pid': problem.pid,
+        'difficulty': problem.difficulty
+    })
+
+# Delete a problem from the database
+@app.route('/api/problems/delete', methods=['POST'])
+@login_required
+def delete_problem():
+    """Delete a specified problem in the database and data folder"""
+    # Admin check
+    if not current_user.admin == 1:
+        return serve_error('You must be an admin to delete a problem',
+                           response_code=401)
+
+    if not request.form['pid']:
+        return serve_error('You must specify a problem to delete',
+                           response_code=400)
+
+    # Delete from problem_data table first to satisfy foreign key constraint
+    problem_data = session.query(ProblemData).\
+        filter(ProblemData.pid == request.form['pid'])
+    if not problem_data.first():
+        return serve_error('Could not find problem data with pid ' +
+                           request.form['pid'], response_code=401)
+    problem_data.delete()
+
+    # Delete any and all sample cases associated w/ problem
+    for case in session.query(SampleCase).\
+            filter(SampleCase.pid == request.form['pid']).all():
+        session.delete(case)
+
+    # Delete from problem table
+    problem = session.query(Problem).\
+        filter(Problem.pid == request.form['pid'])
+    if not problem.first():
+        return serve_error('Could not find problem with pid ' +
+                           request.form['pid'], response_code=401)
+    problem.delete()
+
+    # Commit changes
+    session.flush()
+    session.commit()
+
+    # Delete judge data
+    directory = os.path.join(app.config['DATA_FOLDER'],
+                             'problems', str(request.form['pid']))
+    rmtree(directory)
+
+    return serve_response({
+        'success': True,
+        'deleted_pid': request.form['pid']
+    })
+
+# Update a problem in the database
+@app.route('/api/problems/edit', methods=['POST'])
```

Comment:
This should also be using a different HTTP method. Do 

    @app.route('/api/problems/<pid>', methods=['PUT'])